### ISSUE TITLE
typedef redefinition with different types

### ISSUE THREAD

**Issue by Pitel** [bytedeco/javacpp#186]
I'm trying to build [Opus](https://opus-codec.org) for Android and I'm getting the following error:
```
In file included from /home/pitel/project/app/src/main/jni/jniopus.cpp:26:
/home/pitel/Android/Sdk/ndk-bundle/sysroot/usr/include/jni.h:36:18: error: typedef redefinition with different types ('int64_t' (aka 'long') vs 'long long')
typedef int64_t  jlong;    /* signed 64 bits */
                 ^
/home/pitel/project/app/src/main/jni/jniopus.cpp:22:23: note: previous definition is here
    typedef long long jlong;
                      ^
```

It's happening with NDK r15-beta2, r13b is ok, r14b is uknown, beacause the build fails somewhere in opus.

**Comment by saudet** [comment]
Let's see, could you tell me what you get if you remove these lines from the generated .cpp file?
```cpp
#elif defined(__GNUC__)
    #define _JAVASOFT_JNI_MD_H_

    #define JNIEXPORT __attribute__((visibility("default")))
    #define JNIIMPORT
    #define JNICALL

    typedef int jint;
    typedef long long jlong;
    typedef signed char jbyte;
```

**Comment by Pitel** [comment]
It finishes ok.

**Comment by saudet** [comment]
This is for 32-bit platforms right? `int64_t` gets defined as `long long` on those, so that works, but on 64-bit platforms, it's most likely `long`, and then that's going to be problematic...

**Comment by Pitel** [comment]
I build it for `arm64-v8a`, `armeabi-v7a`, `armeabi`, `x86` and `x86_64`. I'll try running it on Nexus 5X phone, which should be `arm64`.

**Comment by Pitel** [comment]
It works on both 32bit and 64bit ARMs.

**Comment by saudet** [comment]
Great, thanks for testing! The commit above includes the fix.

**Comment by saudet** [comment]
Fix released in version 1.3.3. Thanks for reporting!
